packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -156,29 +215,39 @@ class DestinationFromServiceRetriever {
           'You have provided the `userJwt` and `iss` options to fetch the destination. This is most likely unintentional. Ignoring `iss`.'
         );
       }
-      const encoded = await serviceToken('destination', {
-        ...options
-      });
+
+      if (isXsuaaToken(decodeJwtComplete(options.jwt))) {
+        await verifyJwt(options.jwt, options);
+        const serviceJwtEncoded = await serviceToken('destination', {
+          ...options,
+          jwt: options.jwt
+        });
+        return {
+          type: 'xsuaa',
+          userJwt: getJwtPair(options.jwt),
+          serviceJwt: getJwtPair(serviceJwtEncoded)
+        };
+      }
       return {
-        userJwt: {
-          decoded: await verifyJwt(options.jwt, options),
-          encoded: options.jwt
-        },
-        serviceJwt: { encoded, decoded: decodeJwt(encoded) }
+        type: 'custom',
+        userJwt: getJwtPair(options.jwt),
+        serviceJwt: undefined
       };
     }
 
     if (options.iss) {
       logger.debug(
         'Using `iss` option to fetch a destination instead of a full JWT. No validation is performed.'
       );
-      const payload = { iss: options.iss };
-      const encoded = await serviceToken('destination', {
+      const serviceJwtEncoded = await serviceToken('destination', {
         ...options,
-        jwt: payload
+        jwt: { iss: options.iss }
       });
-      const clientCertJwt = { encoded, decoded: decodeJwt(encoded) };
-      return { serviceJwt: clientCertJwt };
+      return {
+        serviceJwt: getJwtPair(serviceJwtEncoded),
+        type: 'iss',
+        userJwt: undefined
```

Comment:
[pp] Totally irrelevant, I just like it more:

```suggestion
        type: 'iss',
        userJwt: undefined,
        serviceJwt: getJwtPair(serviceJwtEncoded)
```